pystac/utils.py

```diff
@@ -61,27 +138,78 @@ def make_relative_href(
         parent, then source_href will be returned unchanged.
     """
 
-    parsed_source = _urlparse(source_href)
-    parsed_start = _urlparse(start_href)
+    parsed_source = safe_urlparse(source_href)
+    parsed_start = safe_urlparse(start_href)
     if not (
         parsed_source.scheme == parsed_start.scheme
         and parsed_source.netloc == parsed_start.netloc
     ):
         return source_href
 
-    is_path = parsed_start.scheme == ""
-
-    if start_is_dir:
-        start_dir = parsed_start.path
+    if JoinType.from_parsed_uri(parsed_start) == JoinType.PATH:
+        return _make_relative_href_path(parsed_source, parsed_start, start_is_dir)
     else:
-        start_dir = _pathlib.dirname(parsed_start.path)
-    relpath = _pathlib.relpath(parsed_source.path, start_dir)
-    if not is_path:
-        relpath = relpath.replace("\\", "/")
-    if not relpath.startswith("."):
-        relpath = _join(is_path, ".", relpath)
+        return _make_relative_href_url(parsed_source, parsed_start, start_is_dir)
 
-    return relpath
+
+def _make_absolute_href_url(
+    parsed_source: URLParseResult,
+    parsed_start: URLParseResult,
+    start_is_dir: bool = False,
+) -> str:
+
+    # If the source is already absolute, just return it
+    if parsed_source.scheme != "":
+        return urlunparse((parsed_source))
```

Comment:
There's an extra pair of brackets here. Ditto in at least one other place below.

Reply:
Fixed in c48c15bf8537eb81dbf28003eb225806bf55d3ff